src/write.c

```diff
@@ -479,11 +476,8 @@ static avifBool avifEncoderDetectChanges(const avifEncoder * encoder, avifEncode
     if (encoder->data->lastTileColsLog2 != encoder->data->tileColsLog2) {
         *encoderChanges |= AVIF_ENCODER_CHANGE_TILE_COLS_LOG2;
     }
-    if (memcmp(&lastEncoder->widthScale, &encoder->widthScale, sizeof(avifScalingMode)) != 0) {
-        *encoderChanges |= AVIF_ENCODER_CHANGE_WIDTH_SCALE;
-    }
-    if (memcmp(&lastEncoder->heightScale, &encoder->heightScale, sizeof(avifScalingMode)) != 0) {
-        *encoderChanges |= AVIF_ENCODER_CHANGE_HEIGHT_SCALE;
+    if (memcmp(&lastEncoder->scaleMode, &encoder->scaleMode, sizeof(avifScalingMode)) != 0) {
```

Comment:
Note: This is a false positive in case `lastEncoder->scaleMode = { {1, 1}, {1, 1} }` and `lastEncoder->scaleMode = { {2, 2}, {3, 3} }` but this is not important, leave it as is.